humility-core/src/hubris.rs

```diff
@@ -401,7 +401,13 @@ pub struct HubrisFlashMap {
 }
 
 impl HubrisFlashMap {
+    /// # Errors
+    ///
+    /// This will fail if the `HubrisArchive` is fake, i.e. contains zero bytes.
     pub fn new(hubris: &HubrisArchive) -> Result<Self> {
+        if hubris.archive().is_empty() {
+            bail!("archive is required for network use but was not provided");
+        }
```

Comment:
I always forget about that macro. In this case, since rustfmt will wrap it anyway, I'm inclined to leave it as is, but thanks for the reminder that it exists.